design/api.md

```diff
@@ -133,51 +124,38 @@ Status 409 Conflict and `{ success: false }`, if the username is already taken.
 
 ---
 
-Change the account's username.
+### GET `/api/v0/capabilities`
 
+This returns the set of UCANs in UCAN chains that have the authorization UCAN's resource DID as final audience.
 
-### POST `/api/v0/fetch-capabilities/:did`
+This way agents can discover new capabilities, or make use of the server as persistent storage for their UCANs, e.g. when their storage gets wiped, but they manage to keep or recover their DID.
 
----
+In some cases this can be useful for logging in a new devices. E.g. when passkeys are available and are synced between devices, e.g. it's easy to ask a synced passkey to sign a UCAN giving the `"ucan:*": { "*": [{}] }` capability. However, this UCAN on its own doesn't give the account access to anything yet. It can use this endpoint to fetch the `"did:key:<account>": { "*": [{}] }` capability UCANs that delegate to the passkey.
 
-**Authentication**:
-
-```json
-{
-  "iss": "did:<from the path>",
-  "aud": "did:<server DID>",
-  "cap": {}, // Note: Should we introduce a capability? "did:key:zABC": { "capability/fetch": [{}] }?
-  "nnc": "<randomly generated>"
-}
-```
-
-**Request**: `{ ucans: string[] }`
-
-**Response**: `{ ucans: string[] }`
-
----
+In the future this endpoint will be useful for fetching new capabilities from other services that were delegated to a user's account.
 
-This returns a set of UCANs, that, combined with the UCAN from the request body will give the given DID access to new capabilities.
+**Authorization**: UCAN with ability `capability/find`.
```

Comment:
How would this work in the Passkey case? You need to get that capability to start, which seems pretty meta, no?

Also small suggestion: `find` is kind of a strange word here (to me). Perhaps `fetch`?

Reply:
I think the misunderstanding here is that the root issuer you need for the `capability/find` ability *does not need to be the account DID*. It's the device's DID which needs to be the root issuer.

I.e. in the canonical case, when I have `did:key:passkey` I create a UCAN that signs `"did:key:passkey": "capabilities/find"` and do `GET /api/v0/capabilities`. I get back all the "reverse dependencies" of `did:key:passkey` in the UCAN network that the server knows about. In the usual case that'd be one UCAN `iss: did:key:server, aud: did:key:passkey`, because `audience == request UCAN root capability`, and also another UCAN `iss: did:key:account, aud: did:key:server`, because `audience == another UCAN's issuer returned in this request` (and it is needed to further delegate a returned capability).

---

`find` vs. `fetch`? Hm. Yeah. I've been going back and forth on these. Good to have signal on that :)